src/Utils.jl

```diff
@@ -304,6 +305,89 @@ function split_by_season(dates::AbstractArray{<:Dates.DateTime})
     return (MAM, JJA, SON, DJF)
 end
 
+"""
+    split_by_season_across_time(dates::AbstractArray{<:Dates.DateTime})
+
+Split `dates` into a vector of vectors of dates split by seasons across time.
+
+Examples
+=========
+
+```jldoctest
+julia> import Dates
+
+julia> dates = collect(Dates.DateTime(2010, i) for i in 1:12);
+
+julia> split_by_season_across_time(dates)
+5-element Vector{Vector{DateTime}}:
+ [DateTime("2010-01-01T00:00:00"), DateTime("2010-02-01T00:00:00")]
+ [DateTime("2010-03-01T00:00:00"), DateTime("2010-04-01T00:00:00"), DateTime("2010-05-01T00:00:00")]
+ [DateTime("2010-06-01T00:00:00"), DateTime("2010-07-01T00:00:00"), DateTime("2010-08-01T00:00:00")]
+ [DateTime("2010-09-01T00:00:00"), DateTime("2010-10-01T00:00:00"), DateTime("2010-11-01T00:00:00")]
+ [DateTime("2010-12-01T00:00:00")]
+"""
+function split_by_season_across_time(dates::AbstractArray{<:Dates.DateTime})
+    # Dates is not necessarily sorted
+    dates = sort(dates)
+
+    # Empty case
+    if length(dates) == 0
+        return Vector{Vector{eltype(dates)}}[]
+    end
```

Comment:
```suggestion
    isempty(dates) && return Vector{Vector{eltype(dates)}}[]
```
(Or in general, use `isempty` instead of checking the length so that it is a little bit more clear)